api/src/services/users/create_api_key.py

```diff
@@ -46,6 +53,24 @@ def create_api_key(db_session: db.Session, user_id: UUID, json_data: dict) -> Us
     )
     db_session.add(api_key)
 
+    # Import the API key to AWS API Gateway
+    try:
+        _import_api_key_to_aws_gateway(api_key)
+    except Exception as e:
+        # Remove the API key from the database since AWS API Gateway integration failed
+        # This ensures consistency - if the key can't be created in API Gateway,
+        # it shouldn't exist in our database either
+        db_session.rollback()
+        logger.error(
+            "Failed to import API key to AWS API Gateway, rolling back database transaction",
+            extra={
+                "api_key_id": api_key.api_key_id,
+                "key_name": key_name,
+                "error": str(e),
+            },
+        )
```

Comment:
I think we don't need this try/except or log message. If it fails, let the error propagate up, we already logged a more specific error in the API gateway code / will log the error message when the API itself catches in to construct the error response.